### PR TITLE
Add banner to study page if study is private and user does not have a…

### PR DESCRIPTION
**Pull request by dmfalke**
…ccess

closes https://github.com/VEuPathDB/web-eda/issues/809

### CODE REVIEW

**Comment by dmfalke** [comment]
**TODO** update package version

**Comment by dmfalke** [comment]
I also upgraded **@veupathdb** packages since eda was falling behind a minor version.

**Review by jtlong3rd — COMMENTED** [review]
Almost good to merge!

File: Client/src/component-wrappers/StudyRecordHeading.jsx [hunk 1]
```diff
@@ -99,7 +113,28 @@ function mapStateToProps(state) {
...
+  return access === 'private' && !isUserFullyApprovedForStudy(permissions, user.properties.approvedStudies, studyId);
+}
+
+function UserLink({ to, user, children }) {
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const dest = to ? history.createHref(parsePath(to)) : window.location.href;
+  if (user.isGuest) {
+    return (
+      <button type="button" className="link" onClick={() => dispatch(showLoginForm(dest))}>
```
[the remark this answers is not in the document]
Comment (dmfalke) [reply Client/src/component-wrappers/StudyRecordHeading.jsx:132]: I suppose `history.createHref` does not include the origin. Thanks for catching this!

File: Client/src/component-wrappers/StudyRecordHeading.jsx [hunk 2]
```diff
@@ -99,7 +113,28 @@ function mapStateToProps(state) {
...
+        {children}
+      </button>
+    );
+  }
+  return (
+    <Link to={route}>{children}</Link>
```
[the remark this answers is not in the document]
Comment (dmfalke) [reply Client/src/component-wrappers/StudyRecordHeading.jsx:138]: Ah, that should be `to`, not `route`!

**Pull request merged**